policy/src/main/java/org/apache/brooklyn/policy/ha/ElectPrimaryPolicy.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.policy.ha;
```

Comment:
I lean towards putting these in a sub-package, such as `org.apache.brooklyn.policy.ha.failover`. There are several classes that all work together, and that are unrelated to the other ha policies (such as `ServiceReplacer` etc).

Reply:
i've gone with slightly simpler `org.apache.brooklyn.policy.failover` - autoscaling is also HA ... our HA package maybe should be renamed, or it's just simple non-specific HA policies